actors/miner/src/lib.rs

```diff
@@ -5144,6 +5185,35 @@ fn verify_aggregate_seal(
     .context_code(ExitCode::USR_ILLEGAL_ARGUMENT, "aggregate seal verify failed")
 }
 
+fn validate_sealer_id_numbers(
+    rt: &impl Runtime,
+    sealer_id_actor: ActorID,
+    sealer_id_verifier_signature: Vec<u8>,
+    sector_numbers: &BitField,
+) -> Result<(), ActorError> {
+    let params = ext::sealer::ActivateSectorParams {
+        sector_numbers: sector_numbers.clone(),
+        verifier_signature: sealer_id_verifier_signature,
+    };
+
+    let result = rt.send_simple(
```

Comment:
I think we have an issue without running `assert_eq(rt.get_code_cid_for_type(Type::Sealer), rt.get_actor_code_cid(sealer_id_actor))`.  If I'm not mistaken miner actor is trusting sealer actor code to make repeated claims impossible.  Otherwise I could make a new sealer actor on the evm that doesn't disallow repeats  and then claim the same sector in X different messages to reduce my sealing cost by X.

This is only practically exploitable because ACTIVATE_SECTORS_METHOD is in user space method range.  But I think this is ok.  We should deal with trusted code head on by checking actor's code cid and then keep the user space method.

Reply:
Oh yeah definitely